lib/fizzy/stack.hpp

```diff
@@ -43,7 +43,7 @@ class Stack
 
     T& top() noexcept { return m_container.back(); }
 
-    void shrink(size_t new_size) noexcept
+    void shrink(size_t new_size)
     {
         assert(new_size <= size());
         m_container.resize(new_size);
```

Comment:
This was found by clang-tidy as it complained in `mark_frame_unreachable`.

It [turned out](https://en.cppreference.com/w/cpp/container/vector/resize) the `resize` here can be either `constexpr` or regular exception based:
> In overload (1), if T's move constructor is not noexcept and T is not CopyInsertable into *this, vector will use the throwing move constructor. If it throws, the guarantee is waived and the effects are unspecified.

Easier removing the specifier, and this means `mark_frame_unreachable` must have it removed too. Fortunately it it changes nothing.

Reply:
For the real solution I think we'd need `static_assert` that move constructor is noexcept and `OperandStackType` would need to be changed from `enum class`.

Reply:
I think this is not correct. We know that the new size is smaller than the current size. Then there will be no allocation and no move constructors. There will be destructors but they are trivial and don't throw. That's why it is safe to have `noexcept` here. Although `clang-tidy` will not shut up about it because `.resize()` is not `noexcept`. I think better change is to add `//NOLINT` comment.

Reply:
Well technically `resize` is not marked `noexcept`, unless `constexpr` 100% implies that. However we are running C++17 and not C++20.

Reply:
Just realised my first attempt 2 weeks ago was exactly that, disabling it on the line: 7059459b5d72a39b387065a1f470cc57bbea3b9d